### ISSUE TITLE
Support pbkdf2_hmac

### ISSUE THREAD

**Issue by remram44** [remram44/taguette#4]
`hashlib.pbkdf2_hmac()` ([doc](https://docs.python.org/3/library/hashlib.html#hashlib.pbkdf2_hmac)) has been added to the Python standard library. This should probably be supported in addition/replacement of bcrypt.

This shouldn't break any existing database, just an additional method should be added to [User#set_password()](https://github.com/remram44/taguette/blob/80bdbf7571c9191e7e9ebd472aefd519bf428a89/taguette/database.py#L69) and [User#check_password()](https://github.com/remram44/taguette/blob/80bdbf7571c9191e7e9ebd472aefd519bf428a89/taguette/database.py#L77)

See also [#143 on GitLab](https://gitlab.com/remram44/taguette/issues/143)

**Comment by stevendlander** [comment]
Hello!

I will have a PR submitted Oct 1 for this, could you answer/clarify a few things prior to that?

1. Since you want to maintain compatibility with existing databases, _complete removal_ of `bcrypt` is not possible, is this a correct assumption?
2. Would you like this new crypt scheme to be the _new default_ scheme that `User#.set_password()` implements, as dictated by the keyword argument?
3. If yes to 2 above, should `User#.set_password()` _only_ support this new crypt scheme and not allow for _future storage_ of `bcrypt` hashes?  `User#.check_password()` would still be able to check the validity of existing `bcrypt` stored hashes.
2. Unlike `bcrypt`, `hashlib.pbkdf2_hmac` doesn't have a helper function that easily checks passwords.  Is it reasonable/acceptable to store the salt value with the hash value (see below).

```python
# Format to store hashed password with the salt that generated it
self.hashed_password = 'pbkdf2:%s:%s` % (salt, hash)
```

Thanks 👍

**Comment by remram44** [comment]
Definitely wait for October 1 before opening the PR, or it won't count towards your hacktoberfest score (and your free tee-shirt :wink:)

I think pbkdf2 should be the new default for `set_password()`. I would like to keep bcrypt both in `check_password()` (so existing users can log in, as you pointed out) and in `set_password()` (for use in tests).

Storing the salt separated with a colon is reasonable. If the salt is random bytes, you could [base64-encode it](https://docs.python.org/3/library/base64.html#base64.b64encode) so that it is safe for the database, and to make sure it doesn't contain a colon? Just a thought, you might have a better idea.

**Comment by stevendlander** [comment]
Yup, the PR uses `base64` for encoding of the salt _and_ the resulting hash provided by the `pbkdf2_hmac()` function.

I will update the keyword argument in my branch and ensure tests cover the new functionality.

**Comment by remram44** [comment]
Closed by c2bb0c25.
